decidim-pages/app/views/decidim/pages/application/show.html.erb

```diff
@@ -8,7 +8,7 @@
   <div class="row">
     <div class="columns medium-7 mediumlarge-8">
       <div class="section">
-        <%== translated_attribute @page.body %>
+        <%= sanitize translated_attribute @page.body %>
```

Comment:
I'm not sure about this one. @josepjaume @oriolgual @xabier should we sanitize the HTML of the page body?

Reply:
I don't think so, as page bodies are generated by admins and might want to include tags.

Reply:
We could allow style tags here if necessary through some customization of the sanitize call (basically adding `style` to the whitelist for this special case).
But I might be missing something here. How could you enter JS or style tags using the JS editor?

Regarding scripts - like I mentioned above, I already consider the `header_snippets` a bad thing that should not be in this part of the application. Especially since Decidim is always embedded in a custom Rails app and never installed and used out of the box, there is always room to add custom scripts as part of the app. What's the use case for letting administrators inject custom javascript on a page level? Shouldn't something like this always be done by somebody with actual access to the codebase?

Reply:
The idea behind the current implementation of Decidim is that you can have multiple organizations in a single installation. With this setup, we think organization admins shouldn't need to bother system admins for them to include some JS snippet.

Reply:
I only see two roles - 'User Manager' and 'Admin' (looking at 0.6-stable). So basically anybody who can do more than act on behalf of managed users can inject Javascript? I'd really like to see that more restricted.

The other part of my question still stands - I do not see the UI where I, as an admin, could actually enter a script tag without resorting to `curl` or turning JS off and fiddling with hidden field values in the browser.

Reply:
@jkraemer there's currently no way to do that, so we could as well restrict it. The idea is to provide a way to create a page via both an editor and pure code, as an escape hatch to situations where you'd want to add arbitrary content.

As for the roles, there's two major roles: `Admin` and `UserManager`, but there's also the `ProcessAdmin`, which is widely used, which doesn't have these kind of permissions.

As @mrcasals says, although you have full control via code, `decidim` is multi-tenant - we kept being asked for a way to provide JS tracking scripts when needed. But if your concern is the user's being able to inject arbitrary JS, maybe we could provide a tighter integration with Piwik or Google Analytics so the user only needs to enter an ID, thus avoiding this concern.

What do you think?

Reply:
Ah I see, the ProcessAdmin is an Administrator restricted to a single process. I'm still getting to know the whole system while already bending it to our needs...

I get the point about the multi tenancy. But why tracking, and especially tracking through third parties, would be desired in a place where people vote (assuming who voted for what should be private) is beyond me. Guess there's use cases for Decidim I don't think of right now where it's not that much of a concern.

So I'm fine with the `header_snippets` being the only place where JS is allowed to be entered and this only being accessible by Admins. I'll still remove it in my fork though ;)

Guess the specialized UI for just entering the tracking ID is a nice to have, but I'm quite sure there would soon be another use case / different tracking system which would require reintroduction of the raw header snippet text box. Or it would be a constant struggle to support yet another analytics tool ;)

Reply:
Your concerns are perfectly valid, but it's something we've needed in the past and probably didn't put enough thought in it. Maybe we could encourage tracking only in the backend side (in public pages, disassociating the user) or maybe prevent tracking in sensible pages (via not rendering the tracking code) although that might be misleading.

@jkraemer we would really prefer if we could cover your scenario somehow instead of using a fork, as it's the whole point of decidim being modular. Would it work if we provided a setting in `config/initializers/decidim.rb`, say, `allow_header_snippets = false` that would disable tracking altogether?

Reply:
It could be disabled by default so we would actually be encouraging users not to use tracking. Heck, we might even put a comment just above explaining why.

Reply:
@josepjaume great idea about the setting. I'll see if I manage to come up with another PR for that.

Reply:
@josepjaume @jkraemer  nice points!

> Maybe we could encourage tracking only in the backend side (in public pages, disassociating the user) or maybe prevent tracking in sensible pages (via not rendering the tracking code) although that might be misleading.

This follows the golden rule of: Maximum transparency for admins, maximum privacy for users (unless they voluntarily resign for research purposes, etc.)